test/common/conn_pool/conn_pool_base_test.cc

```diff
@@ -92,6 +93,45 @@ class ConnPoolImplBaseTest : public testing::Test {
   std::vector<TestActiveClient*> clients_;
 };
 
+class ConnPoolImplDispatcherBaseTest : public testing::Test {
+public:
+  ConnPoolImplDispatcherBaseTest()
+      : api_(Api::createApiForTest(time_system_)),
+        dispatcher_(api_->allocateDispatcher("test_thread")),
+        pool_(host_, Upstream::ResourcePriority::Default, *dispatcher_, nullptr, nullptr, state_) {
+    // Default connections to 1024 because the tests shouldn't be relying on the
+    // connection resource limit for most tests.
+    cluster_->resetResourceManager(1024, 1024, 1024, 1, 1);
+    ON_CALL(pool_, instantiateActiveClient).WillByDefault(Invoke([&]() -> ActiveClientPtr {
+      auto ret =
+          std::make_unique<NiceMock<TestActiveClient>>(pool_, stream_limit_, concurrent_streams_);
+      clients_.push_back(ret.get());
+      ret->real_host_description_ = descr_;
+      return ret;
+    }));
+    ON_CALL(pool_, onPoolReady(_, _))
+        .WillByDefault(Invoke([](ActiveClient& client, AttachContext&) -> void {
+          ++(reinterpret_cast<TestActiveClient*>(&client)->active_streams_);
```

Comment:
any reason not to dynamic_cast & ASSERT_TRUE here?

Reply:
maybe just factor this out until a new function `incrementActiveStreams(&client)` and you can do the dynamic-cast and ASSERT in there?